Week 2 Section 2 Added

GaxysGalactic opened the pull request:
Spotify Project is now complete!

# Debug Testing

![E3](https://user-images.githubusercontent.com/44332093/179067550-b09a323a-3d11-44fd-a2d1-9209323a1e9f.PNG)

# MainMenu

![MainMenu](https://user-images.githubusercontent.com/44332093/179067584-0ae6ad94-8b01-4559-82d2-c453a398211c.PNG)

# Create Playlist

![CreatePlaylist1](https://user-images.githubusercontent.com/44332093/179067636-bf6d1459-50f7-4904-9700-3e9de5e5f5e1.PNG)
![CreatePlaylist2](https://user-images.githubusercontent.com/44332093/179067639-44a1eda5-14b8-41ed-a145-f3128134eb75.PNG)

# Browse Playlists

![BrowsePlaylists](https://user-images.githubusercontent.com/44332093/179067678-d2190237-5ae6-4a78-bf15-58b8507005c0.PNG)
![BrowsePlaylists2](https://user-images.githubusercontent.com/44332093/179067682-b9091954-0421-4d6c-aea7-98ccd34dc761.PNG)

# Input Validation

![InputValidation](https://user-images.githubusercontent.com/44332093/179067723-1ad257cb-6b72-4478-a908-57409ca422e9.PNG)
![InputValidation2](https://user-images.githubusercontent.com/44332093/179067727-f2993da9-96aa-49f7-84f4-d9b143fab01b.PNG)

GaxysGalactic committed:
Week 2 Section 2 Added

Spotify Project is now complete. I will be adding the screenshots of the results to this description later.

JDAlzate commented on Month1/Week2/Section2/SpotifyProject/Song.h:
```diff
@@ -0,0 +1,54 @@
+
+	void setDuration(int pDuration);
+
+	std::string getSongTitle();
```
Some of these methods could be declared const 👀 Remember the examples won't use const, but you're supposed to maintain const correctness.

JDAlzate commented on Month1/Week2/Section2/SpotifyProject/Song.cpp:
```diff
@@ -0,0 +1,81 @@
+
+bool Song::isValid()
+{
+	return duration < 360;
```
Is no song allowed to be more than 6 minutes ever?

JDAlzate commented on Month1/Week2/Section2/SpotifyProject/Playlist.cpp:
```diff
@@ -0,0 +1,60 @@
+		songCount++;
+		return true;
+	}
+	else
```
This else statement is not wrong but is redundant. Your if statement returns true and anything else should return false. So you could simply write
```
if (condition)
{
    // Do stuff
    return true;
}

return false;
```

JDAlzate reviewed (changes requested):
Please make a commit for each exercise you complete :)
Please fix the const correctness for the whole project.

GaxysGalactic committed:
Const Correctness, Redundancy, Validity

- [FIX] Redundant else statement in playlist
- [FIX] Const Correctness for classes
- [NEW]  Validation of a song makes more sense

JDAlzate commented on Month1/Week2/Section2/SpotifyProject/Playlist.cpp:
```diff
@@ -21,7 +21,7 @@ void Playlist::setPlaylistTitle(std::string pPlaylistTitle)
 }
 
-std::string Playlist::getPlaylistTitle()
+const std::string Playlist::getPlaylistTitle()
```
const before the function doesn't make the function const. It is actually returning a const value, which is actually not good practice hehe. To make a method const, the const keyword should be placed _after_ the method. So:
``` std::string Playlist::getPlaylistTitle() const```

JDAlzate reviewed (changes requested):
Issue with const correctness.

GaxysGalactic committed:
Refixed Const Correctness

- [FIX] Fixed the methods so that the keyword "const" is actually placed in the correct place.

JDAlzate reviewed (approved):
Nice! Merge merge merge
